apps/css-workshop/src/components/Checkbox.astro

```diff
@@ -0,0 +1,71 @@
+---
+type Props = {
+  isLoading?: boolean;
+  indeterminate?: boolean;
+  checked?: boolean;
+  disabled?: boolean;
+  variant?: 'eyeball';
+  basic?: boolean;
+  status?: 'negative' | 'positive' | 'warning';
+} & astroHTML.JSX.HTMLAttributes;
+
+const { status, isLoading, indeterminate, disabled, variant, basic, ...props } = Astro.props;
+---
+
+{
+  basic && (
+    <input
+      class:list={[
+        'iui-checkbox',
+        {
+          'iui-loading': isLoading,
+          'iui-checkbox-visibility': variant?.includes('eyeball'),
+        },
+      ]}
+      type='checkbox'
+      name={indeterminate ? 'indeterminate' : undefined}
+      disabled={disabled}
+      {...props}
+    />
+  )
+}
+{
+  !basic && (
+    <label
+      class:list={[
+        'iui-checkbox-wrapper',
+        {
+          'iui-disabled': disabled,
+          'iui-loading': isLoading,
+          [`iui-${status}`]: !!status,
+        },
+      ]}
+      {...props}
```

Comment:
This one spreads `...props` in two different places. It should only spread on the input.

Reply:
resolved in new branch